From 58fcd34c12dcd8c529740b10cd9722337b6314d1 Mon Sep 17 00:00:00 2001
From: Mike Snitzer <snitzer@redhat.com>
Date: Fri, 13 Jun 2014 14:36:50 -0400
Subject: [block] blk-exec: Cleaning up local variable address returned

Message-id: <1402670238-13916-123-git-send-email-snitzer@redhat.com>
Patchwork-id: 83906
O-Subject: [RHEL7.1 PATCH 122/150] block: blk-exec.c: Cleaning up local variable address returnd
Bugzilla: 1105204
RH-Acked-by: David Milburn <dmilburn@redhat.com>
RH-Acked-by: Jeff Moyer <jmoyer@redhat.com>

BZ: 1105204

Upstream commit de83953f9d710f84c4a162a1d498a73475c07d98
Author: Rickard Strandqvist <rickard_strandqvist@spectrumdigital.se>
Date:   Sat Jun 7 00:37:26 2014 +0200

    block: blk-exec.c: Cleaning up local variable address returnd

    Address of local variable assigned to a function parameter

    This was partly found using a static code analysis program called cppcheck.

    Signed-off-by: Rickard Strandqvist <rickard_strandqvist@spectrumdigital.se>
    Signed-off-by: Jens Axboe <axboe@fb.com>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/block/blk-exec.c b/block/blk-exec.c
index dbf4502..f4d27b1 100644
--- a/block/blk-exec.c
+++ b/block/blk-exec.c
@@ -132,6 +132,11 @@ int blk_execute_rq(struct request_queue *q, struct gendisk *bd_disk,
  if (rq->errors)
   err = -EIO;
 
+ if (rq->sense == sense) {
+  rq->sense = NULL;
+  rq->sense_len = 0;
+ }
+
  return err;
 }
 EXPORT_SYMBOL(blk_execute_rq);
-- 
1.7.1